api/src/main/java/org/apache/iceberg/types/GetProjectedIds.java

```diff
@@ -39,9 +39,7 @@
 
   @Override
   public Set<Integer> field(Types.NestedField field, Set<Integer> fieldResult) {
-    if (fieldResult == null) {
-      fieldIds.add(field.fieldId());
-    }
+    fieldIds.add(field.fieldId());
```

Comment:
This is related to [my comment on the other PR](https://github.com/apache/iceberg/pull/2952/files#r685370873). I don't think that we should return inner IDs other than struct IDs or else it isn't clear how lists and maps should be handled.